dev_guide/new_app.adoc

```diff
@@ -39,51 +40,155 @@ application is created.
+specification) can be specified along with the repository using a `~` as a separator.
+
+.Use image stream myproject/my-ruby to build the source at a remote github repository
 ====
 ----
-$ oc new-app mysql
+$ oc new-app myproject/my-ruby~https://github.com/openshift/ruby-hello-world.git
+----
+====
+
+.Use Docker image openshift/ruby-20-centos7:latest to build source in a local repository
+====
+----
+$ oc new-app openshift/ruby-20-centos7:latest~/home/user/code/my-ruby-app
 ----
 ====
 
-To create an application using an image in a private registry, specify the file
-path:
+=== Specifying an image
+
+New-app will generate the necessary artifacts to deploy an existing image as an application. Images can come from image streams in
+the OpenShift server, images in a specific registry or DockerHub, or images in the local Docker server. New-app will attempt to determine
+the type of image specified in the arguments passed to it. However, you can explicitly tell new-app whether the image is a docker image (using the 
+`--docker-image` argument) or an image stream (using the `-i|--image` argument).
+
+
+[NOTE]
+====
+If you specify an image from your local Docker repository, you will need to ensure that the same image is available
+to the OpenShift cluster nodes.
+====
+
+
+
+.Creating an application from the https://registry.hub.docker.com/_/mysql/[DockerHub MySQL Image]:
+====
+----
+$ oc new-app mysql
+----
+====
 
+To create an application using an image in a private registry, specify the full Docker image specification:
+.Creating an application from a local registry
 ----
-$ oc new-app <myregistry.com>/<mycompany>/<image_name>
+$ oc new-app myregistry:5000/example/myimage
 ----
 
-You can also apply a label when creating an application. This is recommended, as
-labels make it easy to collectively select, manipulate, and delete objects
-associated with the application. See the
-link:../cli_reference/basic_cli_operations.html#common-operations[CLI Reference]
-for more details. For example, to apply label `name=hello-world` when creating
-an application from the `ruby-hello-world` remote repository:
+[NOTE]
+====
+If the registry that the image comes from is not secured with SSL, make sure that the Docker daemon on the OpenShift nodes
+is run with the `--insecure-registry` flag pointing to that registry. You also need to tell new-app that the image comes from 
+an insecure registry with the `--insecure-registry=true` flag.
+====
+
+To create an application from an existing 
+link:architecture/core_objects/openshift_model.html#imagestream[image stream], specify the namespace (optional), name, and tag (optional) for the 
+image stream:
 
+.Create an application from an existing image stream, with a specific tag
 ====
 ----
-$ oc new-app https://github.com/openshift/ruby-hello-world -l name=hello-world
+$ oc new-app my-stream:v1
```

Comment:
needs discussion about what project the imagestream is expected to be in and how to specify other projects.